packages/web/src/common/store/pages/signon/sagas.js

```diff
@@ -739,14 +740,7 @@ function* watchConfigureMetaMask() {
 }
 
 function* watchFollowArtists() {
-  while (
-    yield all([
```

Comment:
This while(yield) thing was a killer. If you were signed into an account when you started the session, then the `take(accountActions.fetchAccountSucceeded.type)` requirement was already fulfilled before you start the sign up flow for the new account. So when `signOnActions.FOLLOW_ARTISTS` was dispatched from the follow artists component, `followArtists` would fire right away even though it's supposed to wait until the new account is created and fetched. This is why the `followArtists` step would ultimately fail (since the user account is empty).

Reply:
Simplified this by just having it take that one action and dispatching it when we actually want it to go.